lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -228,9 +228,9 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
                                          "Only Tensor types supported in TOSA");
 
     if (auto lhsElemTy = lhsType.getElementType().dyn_cast<IntegerType>()) {
-      if (lhsElemTy.getWidth() > 32)
+      if (lhsElemTy.getWidth() > 64)
```

Comment:
TOSA doesn't support 64-bit integers. Is this really a 64-bit value, or just defaulting to 64-bit?

Reply:
I trace the ops, looks like we create torch.constant.int to int64_t in ConvertAtenOp<PrimNumToTensorScalarOp>
```
%int0 = torch.constant.int 0
%230 = torch.prim.NumToTensor.Scalar %int0 : !torch.int -> !torch.vtensor<[],si64>
%232 = torch.aten.broadcast_to %230, %231 : !torch.vtensor<[],si64>, !torch.list<int> -> !torch.vtensor<[128,128],si64>
%233 = torch.aten.copy %232, %229, %false : !torch.vtensor<[128,128],si64>, !torch.vtensor<[128,128],si64>, !torch.bool -> !torch.vtensor<[128,128],si64>
%234 = torch.aten.slice.Tensor %233, %int1, %int0, %int9223372036854775807, %int1 : !torch.vtensor<[128,128],si64>, !torch.int, !torch.int, !torch.int, !torch.int -> !torch.vtensor<[128,128],si64>
%235 = torch.aten.unsqueeze %234, %int0 : !torch.vtensor<[128,128],si64>, !torch.int -> !torch.vtensor<[1,128,128],si64>
%678 = torch.aten.unsqueeze %235, %int1 : !torch.vtensor<[1,128,128],si64>, !torch.int -> !torch.vtensor<[1,1,128,128],si64>
%722 = torch.aten.neg %678 : !torch.vtensor<[1,1,128,128],si64> -> !torch.vtensor<[1,1,128,128],si64>
%723 = torch.aten.add.Scalar %722, %int256, %int1 : !torch.vtensor<[1,1,128,128],si64>, !torch.int, !torch.int -> !torch.vtensor<[1,1,128,128],si64>
```

Reply:
Is there any possibility we change tosa to support 64-bit?

Reply:
With TOSA, we're trying to support hardware where si64 either doesn't exist, or is very expensive. Adding support for si64 as a default for torch constant ints would not be desirable.
If we really have to, we could consider adding it to the dialect, and then a validation pass that verifies i32 capability for hardware to use. That's a second choice for me.

Reply:
This is a recurring issue I hear from the IREE developers as well regarding f64/i64 support. Technically, !torch.int and !torch.float are si64 and f64 respectively. So truncating them would require a special user flag to say that they don't care about it, since it is possible to write code that does care. 

I filed an issue about this: https://github.com/llvm/torch-mlir/issues/1615

Reply:
@Svoch

Reply:
I think the cast is okay for now. We've got the torch issue that Sean filed, perhaps we should file one here to note that this needs a long term fix, possibly finding alignment for the groups like IREE that also have this problem.